### DISCUSSION TITLE
Is there a way to show the next days agile pricing in a graph in HA, like what the Octopus portal produces?

### DISCUSSION THREAD

**Discussion by EPicLURcher** [BottlecapDave/HomeAssistant-OctopusEnergy#619] [Q&A] [answered]
Hi,

I'm looking to recreate this view within HA:

![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/74502301/958b3245-8a4b-42dd-969f-db18fae7ecad)

The object that holds all of the next day rates is an event, and I cannot work out how to use this with a sensor graph:
event.octopus_energy_electricity_[redacted]_current_day_rates

this is what the sensor contains: (a sample of the first 2 rates of the day)
event_types: octopus_energy_electricity_current_day_rates
event_type: octopus_energy_electricity_current_day_rates
rates: 
- start: '2023-12-10T00:00:00+00:00'
  end: '2023-12-10T00:30:00+00:00'
  value_inc_vat: 0.091665
  is_capped: false
- start: '2023-12-10T00:30:00+00:00'
  end: '2023-12-10T01:00:00+00:00'
  value_inc_vat: 0.11907
  is_capped: false

Any ideas on how I can use this to create the view I'm after in HA?

**Comment by anothermort** [comment]
I am sure I saw some one using the apex card to do this.

Found it [https://github.com/lozzd/octopus-energy-rates-card/issues/13](url)

**Reply by EPicLURcher** [reply]
thank you!

**Comment by BottlecapDave** [comment, marked as answer]
Hi. As @anothermort mentioned, a lot of users use apex charts. You can see https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy/community/ for inspiration on how this can be used with the data to make a chart similar to what you're after

**Reply by EPicLURcher** [reply]
this is perfect - thanks for the reply

**Comment by firstandlast1966** [comment]
Did anyone manage to get a rates card working? I have the bottlecapdave integration installed but the lozzd link gives a 404.

**Reply by BottlecapDave** [reply]
I've just tried the link and it seems to be working for me?
